Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_fd.nut

```diff
@@ -109,9 +110,29 @@ void function FD_PlayerRespawnCallback( entity player )
 	if( player in file.players )
 		file.players[player].lastRespawn = Time()
 
+	if( GetCurrentPlaylistVarInt( "fd_at_unlimited_ammo", 0 ) )
+		Unlimited_Ammo( player )
+
 	Highlight_SetFriendlyHighlight( player, "sp_friendly_hero" )
 }
 
+void function OnPlayerGetsNewPilotLoadout( entity player, PilotLoadoutDef loadout )
+{
+	if( GetCurrentPlaylistVarInt( "fd_at_unlimited_ammo", 0 ) )
+		Unlimited_Ammo( player )
+}
+
+void function Unlimited_Ammo( entity player )
+{
+	entity weapon = player.GetMainWeapons()[1]
+	if( !weapon.HasMod( "at_unlimited_ammo" ) )
+	{
+		array<string> mods = weapon.GetMods()
+		mods.append( "at_unlimited_ammo" )
+		weapon.SetMods( mods )
+	}
+}
+
```

Comment:
Function name should be more discriptive like GivePlayerInfiniteAntiTitanAmmo.
Also this function only works when the Anti Titan Weapon is in the 2nd slot.

Reply:
1.Rename the function name in new commit.
2.I remember that Anti Titan Weapon always in a fixed position
After weapon dropped ,mod still in the entity weapon;Or use `GetWeaponClassName()` instead of it?

Reply:
Anti Titan Weapons can be in slot 2 or 3.
To test if it is a anti titan weapon you can do `( weapon.GetWeaponType() == WT_ANTITITAN )`

Reply:
you can also use `GetATWeapons(player)` to get the at weapons it return an array with all the anti titan weapons a player has

Reply:
thx,guy.It's a good idea

Reply:
so...can we resolve the conversation?Or still something I need to change